### ISSUE TITLE
Support "project" in "Linter.lint" API

### ISSUE THREAD

**Issue by bennyn** [palantir/tslint#4161]
### Feature request

**Is your feature request related to a problem? Please describe.**
I want to run TSLint from a Gulp task. Therefor I am [using TSLint as a Node.js library](https://palantir.github.io/tslint/usage/library/). Unfortunately I could not find out how to pass a `tsconfig.json` file (`--project`) to the `Linter` to determine which files will be linted. The `Linter.lint` method only seems to support a `configuration`.

**Describe the solution you'd like**
I need an API equivalent to `tslint --project tsconfig.json --config tslint.json`.

The signature of the current `lint` method looks like this:

```js
lint(fileName, source, configuration)
```

It could be overloaded to:

```js
lint(project, configuration)
```

Another good approach could be this:

```js
const linter = new tslint.Linter({
  project: 'tsconfig.json',
  config: 'tslint.json',
  fix: false
});
```

**Describe alternatives you've considered**
At the moment I am doing the following to achieve my tasks (but it involves a huge chunk of code because of the current API all files must be traversed by hand):

```js
const tslint = require('tslint');
const gutil = {
  PluginError: require('plugin-error'),
};

gulp.task('lint:ts', done => {
  const program = tslint.Linter.createProgram(`${process.cwd()}/tsconfig.json`);
  const linter = new tslint.Linter({
    fix: false,
    formatter: "prose",
  }, program);

  const files = tslint.Linter.getFileNames(program);
  files.forEach(fileName => {
    const fileContents = program.getSourceFile(fileName).getFullText();
    const configuration = tslint.Configuration.findConfiguration(`${process.cwd()}/tslint.json`, fileName).results;
    linter.lint(fileName, fileContents, configuration);
  });

  const results = linter.getResult();
  if (results.errorCount === 0) {
    done();
  } else {
    throw new gutil.PluginError('tslint', new Error(results.output));
  }
});
```

**Comment by JoshuaKGoldberg** [comment]
@bennyn `Linter` accepts a `program` in its constructor, then `lint` will use files from the program. Is that what you're looking for? It's unclear to me from your description why that isn't acceptable?

#3797 tracks cleaning this area up.

**Comment by bennyn** [comment]
@JoshuaKGoldberg `program` is exactly what I was looking for. Thanks! I just couldn't find it because the parameter for the TSLint CLI is called `project` (and not `program`).

My code now works okay:

```js
const tslint = require('tslint');
const program = tslint.Linter.createProgram(`${process.cwd()}/tsconfig.json`);
const linter = new tslint.Linter(
  {
    fix: false,
    formatter: 'prose',
  },
  program
);
```
